packages/blade/src/components/Modal/Modal.web.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { useEffect } from 'react';
+import styled, { css, keyframes } from 'styled-components';
+import { FloatingFocusManager, useFloating } from '@floating-ui/react';
+import usePresence from 'use-presence';
+import { ModalPortal } from './ModalPortal';
+import { ModalHeader } from './ModalHeader';
+import type { ModalHeaderProps } from './ModalHeader';
+import { ModalFooter } from './ModalFooter';
+import type { ModalFooterProps } from './ModalFooter';
+import { ModalBody } from './ModalBody';
+import type { ModalBodyProps } from './ModalBody';
+import { ModalContext } from './ModalContext';
+import { ModalBackdrop } from './ModalBackdrop';
+import {
+  modalBorderRadius,
+  modalHighestZIndex,
+  modalMaxHeight,
+  modalMaxWidth,
+  modalMinWidth,
+  modalResponsiveScreenGap,
+} from './modalTokens';
+import {
+  MetaConstants,
+  castWebType,
+  isValidAllowedChildren,
+  makeAccessible,
+  makeMotionTime,
+  makeSize,
+  metaAttribute,
+} from '~utils';
+import { BaseBox } from '~components/Box/BaseBox';
+import { useTheme } from '~components/BladeProvider';
+import { Box } from '~components/Box';
+
+type ModalProps = {
+  /**
+   *  Children of Modal
+   * Only ModalHeader, ModalBody and ModalFooter are allowed as children
+   */
+  children: React.ReactNode;
+  /**
+   Sets the modal to open or close
+   * @default false
+   */
+  isOpen: boolean;
+  /**
+   *  Callback function when user clicks on close button or outside the modal or on pressing escape key.
+   */
+  onDismiss: () => void;
+  /**
+   *  Ref to the element to be focused on opening the modal.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  initialFocusRef?: React.MutableRefObject<any>;
```

Comment:
This eslint ignore might interfere with the jsdoc comment. let's move it to top of the file.